cmd/local-locker.go

```diff
@@ -127,16 +127,17 @@ func (l *localLocker) Unlock(_ context.Context, args dsync.LockArgs) (reply bool
 	l.mutex.Lock()
 	defer l.mutex.Unlock()
 	err = nil
-
 	for _, resource := range args.Resources {
-		if !l.canTakeUnlock(resource) {
-			// Unless it is a write lock reject it.
-			err = fmt.Errorf("unlock attempted on a read locked entity: %s", resource)
-			continue
-		}
-		lri, ok := l.lockMap[resource]
-		if ok {
+		if lri, ok := l.lockMap[resource]; ok {
+			if !l.canTakeUnlock(resource) {
+				// Unless it is a write lock reject it.
+				err = fmt.Errorf("unlock attempted on a read locked entity: %s", resource)
+				continue
+			}
 			reply = l.removeEntry(resource, args, &lri) || reply
+		} else {
+			// return true if lock not exits.
+			reply = true
```

Comment:
This is wrong @yudoutingle you should never return true for Unlocks() on non-existent resources. This shows bugs here is the simplest fix

```diff
diff --git a/cmd/local-locker.go b/cmd/local-locker.go
index f89efcc7c..8eff129df 100644
--- a/cmd/local-locker.go
+++ b/cmd/local-locker.go
@@ -129,12 +129,12 @@ func (l *localLocker) Unlock(_ context.Context, args dsync.LockArgs) (reply bool
        err = nil
 
        for _, resource := range args.Resources {
-               if !l.canTakeUnlock(resource) {
+               lri, ok := l.lockMap[resource]
+               if ok && !isWriteLock(lri) {
                        // Unless it is a write lock reject it.
                        err = fmt.Errorf("unlock attempted on a read locked entity: %s", resource)
                        continue
                }
-               lri, ok := l.lockMap[resource]
                if ok {
                        reply = l.removeEntry(resource, args, &lri) || reply
                }
```